arch/lkl/kernel/proc.c

```diff
@@ -1,69 +1,72 @@
 
+#include <asm/host_ops.h>
 #include <linux/seq_file.h>
+#include <linux/slab.h>
 
 #ifdef CONFIG_PROC_FS
 
-/*
- * TODO: This currently just hard-codes some output for /proc/cpuinfo that is
- * plausible and can be parsed by an application. Note though that CPU features, 
- * the virtual CPU core count and socket count are wrong.
- */
+static int cpuinfo_generated = 0;
```

Comment:
Is this needed, isn't `cpuinfo_buf == NULL` sufficient?

Reply:
Yes, right. Done.